include/IECore/TypeTraits.h

```diff
@@ -298,6 +302,21 @@ template<typename T> struct IsStringVectorTypedData : public boost::false_type {
 template<> struct IsStringVectorTypedData< TypedData<std::vector<std::string> > > : public boost::true_type {};
 template<> struct IsStringVectorTypedData< TypedData<std::vector<IECore::InternedString> > > : public boost::true_type {};
 
+/// IsPathMatcher
+template< typename T > struct IsPathMatcher : public boost::false_type {};
+template<> struct IsPathMatcher< PathMatcher > : public boost::true_type {};
+template<> struct IsPathMatcher< const PathMatcher > : public boost::true_type {};
+
+/// IsPathMatcherTypedData
+template< typename T > struct IsPathMatcherTypedData : boost::mpl::and_< IsTypedData<T>, IsPathMatcher< typename ValueType<T>::type > > {};
+
+/// IsDateTime
+template< typename T > struct IsDateTime : public boost::false_type {};
+template<> struct IsDateTime< boost::posix_time::ptime > : public boost::true_type {};
+template<> struct IsDateTime< const boost::posix_time::ptime > : public boost::true_type {};
+
+/// IsDateTimeTypedData
+template< typename T > struct IsDateTimeTypedData : boost::mpl::and_< IsTypedData<T>, IsDateTime< typename ValueType<T>::type > > {};
```

Comment:
I think it's unlikely that these new traits have enough uses to justify their existence - I don't think they're necessary for the current use case even (see other comment).

Reply:
removed